### DISCUSSION TITLE
Finding assignment operations for a given local symbol

### DISCUSSION THREAD

**Discussion by jnm2** [dotnet/roslyn#51767] [Q&A] [answered]
Given this syntax, and an ILocalReferenceOperation for `queryable` in `return queryable;`, how can I:

- Obtain the IInvocationOperation `enumerable.AsQueryable()`?
- Know that this is the most recent assignment to `queryable` possible? I'm happy to bail if `queryable` is ever reassigned, but I do need a way to know whether or not it is. And if it's easy to loop over all possible most recent assignments, then I want to do that.

```cs
var queryable = enumerable.AsQueryable();
return queryable;
```

`AnalyzeDataFlow` seems to require passing in either a single expression or two statements, but I am not sure how to use it to analyze something like this:

```cs
return enumerable.AsQueryable() is var queryable ? queryable : null;
```

**Comment by jnm2** [comment, marked as answer]
@333fred showed me https://github.com/dotnet/roslyn-analyzers/blob/main/src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/PointsToAnalysis/PointsToAnalysis.cs in https://www.nuget.org/packages/Microsoft.CodeAnalysis.AnalyzerUtilities which is what roslyn-analyzers uses for this analysis.

**Comment by jnm2** [comment]
This looks really interesting: https://github.com/dotnet/roslyn/pull/51898/files

**Reply by ryzngard** [reply]
@jnm2 yea, we're working on a data tracking feature for #25994. Likely will have overlap and/or use https://github.com/dotnet/roslyn-analyzers/blob/main/src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/PointsToAnalysis/PointsToAnalysis.cs
